lib/dialects/mysql/transaction.js

```diff
@@ -30,11 +30,7 @@ Object.assign(Transaction_MySQL.prototype, {
         if (status === 1) t._resolver(value);
         if (status === 2) {
           if (isUndefined(value)) {
-            if (
-              sql &&
-              sql.toUpperCase() === 'ROLLBACK' &&
-              t.doNotRejectOnRollback
-            ) {
+            if (t.doNotRejectOnRollback && /^ROLLBACK\b/i.test(sql)) {
```

Comment:
What happens if sql is nil here? It just resolves to false or throw an error?

Reply:
Returns `false`